### PR TITLE
GI-1 Initial Setup

### LINKED ISSUE #20: GI-1 - Build initial sample map using google maps api
Create a sample map using the following instructions:
https://developers.google.com/maps/documentation/javascript/overview#maps_map_simple-typescript

Additional items:

- [x] insert a search box
- [x] set boundaries for boston

### PR DESCRIPTION
**Pull request by hams7504**
### ℹ️ Issue

Closes #20 

### 📝 Description

Write a short summary of what you added. Why is it important? Any member of C4C should be able to read this and understand your contribution -- not just your team members.

Briefly list the changes made to the code:
1. Added support for this.
2. And removed redunant use of that.
3. Also this was included for reasons.

### ✔️ Verification

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves.

Provide screenshots of any new components, styling changes, or pages. 

### 🏕️ (Optional) Future Work / Notes

Did you notice anything ugly during the course of this ticket? Any bugs, design challenges, or unexpected behavior? Write it down so we can clean it up in a future ticket!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +41,9 @@
     "react-query": "^3.39.3",
     "react-reveal": "^1.2.2",
     "react-router-dom": "6.11.2",
+    "react-scripts": "^5.0.1",
     "react-typical": "^0.1.3",
+    "styled-components": "^6.0.5",
```
Comment (SofieDunt) [inline package.json:46]: ❓ Is this used?

**Review by SofieDunt — COMMENTED** [review]
Great work on this so far - the code looks clean!

File: apps/green-infrastructure/frontend/src/components/map/SampleMap.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    strokeColor: '#50B0E6',
+    strokeOpacity: 1.0,
+    strokeWeight: 3.0,
+    fillColor: '#50B0E6',
+    fillOpacity: 0.3
+  };
+
+  featureLayer.style = (options) => {
+    const feature = options.feature as google.maps.PlaceFeature;
+    if (feature.placeId === 'ChIJGzE9DS1l44kRoOhiASS_fHg') { // Place ID for Boston
```
[the remark this answers is not in the document]
Comment (hams7504) [reply apps/green-infrastructure/frontend/src/components/map/SampleMap.tsx:41]: Done!

File: apps/green-infrastructure/frontend/src/components/mapIcon/MapIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    rotation: number;
+    scale: number;
+  };
+}
+
+export function createMapIcon(
+  name: string,
+  lat: number,
+  lng: number,
+  icon: MapIcon["icon"]
```
[the remark this answers is not in the document]
Comment (hams7504) [reply apps/green-infrastructure/frontend/src/components/mapIcon/MapIcon.tsx:19]: Done! MapIcon now represents the display icons specifically designed for the map and MapPoint now represents a pin in the map.

File: apps/green-infrastructure/frontend/src/components/map/Map.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            map.setCenter(place.geometry.location);
+            map.setZoom(17);
+          }
+
+          marker.setPosition(place.geometry.location);
+          marker.setVisible(true);
+        });
+      });
+    }
+  }, [zoom]);
```
Comment (chromium-52) [inline apps/green-infrastructure/frontend/src/components/map/Map.tsx:125]: :question: Is there a reason why we're creating a new map every time the zoom level changes?

**Review by chromium-52 — COMMENTED** [review]
:100: there's some code cleanup here and there but otherwise, I think you're headed in the right direction!

As a slightly unrelated note, now that I'm thinking about this again, I think it might actually be better to just have green infra have its own repo (basically second-guessing the existence of this repo) so we should also discuss this really soon :thinking_face:

**Review by JamesColesanti — APPROVED** [review]

**Review by SofieDunt — APPROVED** [review]
Approving to deploy

**Review by chromium-52 — DISMISSED** [review]
There's def a bit of refactoring and code cleanup we'd need to do but should be fine as a poc for now. Good work on this!

**Review by chromium-52 — APPROVED** [review]

**Review by mahekagg — APPROVED** [review]

**Review by JamesColesanti — APPROVED** [review]

**Pull request merged**